docker/elasticsearch-setup/ml-model-index-config.json

```diff
@@ -0,0 +1,250 @@
+          },
+          "comma_pattern_ngram": {
+            "filter": [
+              "lowercase",
+              "autocomplete_filter"
+            ],
+            "type": "custom",
+            "tokenizer": "comma_tokenizer"
+          },
+          "delimit": {
+            "filter": [
+              "lowercase",
+              "custom_delimiter"
+            ],
+            "tokenizer": "whitespace"
+          },
+          "ml_model_pattern_ngram": {
+            "filter": [
+              "lowercase",
+              "autocomplete_filter"
+            ],
+            "type": "custom",
+            "tokenizer": "ml_model_pattern"
+          },
+          "custom_browse_slash": {
+            "filter": [
+              "lowercase"
+            ],
+            "type": "custom",
+            "tokenizer": "path_hierarchy"
+          }
+        },
+        "tokenizer": {
+          "path_hierarchy_tokenizer": {
+            "type": "path_hierarchy",
+            "replacement": "/",
+            "delimiter": "."
+          },
+          "custom_ngram": {
+            "type": "ngram",
+            "min_gram": "3",
+            "max_gram": "50"
+          },
+          "slash_tokenizer": {
+            "pattern": "[/]",
+            "type": "pattern"
+          },
+          "comma_tokenizer": {
+            "pattern": ",",
+            "type": "pattern"
+          },
+          "ml_model_pattern": {
+            "pattern": "[./]",
+            "type": "pattern"
+          }
+        }
+      }
+    }
+  },
+  "mappings": {
+    "properties": {
+      "browsePaths": {
+        "type": "text",
+        "fields": {
+          "length": {
+            "type": "token_count",
+            "analyzer": "slash_pattern"
+          }
+        },
+        "analyzer": "custom_browse_slash",
+        "fielddata": true
+      },
+      "origin": {
+        "type": "keyword",
+        "fields": {
+          "ngram": {
+            "type": "text",
+            "analyzer": "custom_ngram"
+          }
+        },
+        "normalizer": "my_normalizer"
+      },
+      "active": {
+        "type": "boolean"
+      },
+      "hasOwners": {
+        "type": "boolean"
+      },
+      "name": {
+        "type": "keyword",
+        "fields": {
+          "model_pattern_ngram": {
+            "type": "text",
+            "analyzer": "model_pattern_ngram"
+          },
+          "delimited": {
+            "type": "text",
+            "analyzer": "delimit"
+          },
+          "ngram": {
+            "type": "text",
+            "analyzer": "custom_ngram"
+          },
+          "pattern": {
+            "type": "text",
+            "analyzer": "model_pattern"
+          }
+        },
+        "normalizer": "my_normalizer"
+      },
+      "owners": {
+        "type": "text",
+        "fields": {
+          "ngram": {
+            "type": "text",
+            "analyzer": "comma_pattern_ngram"
+          }
+        },
+        "analyzer": "comma_pattern"
```

Comment:
why do you need a `comma_pattern` analyzer here? This is used when owners are stored as string, separated by commas. Looks like you are storing `owners` as an array of strings in search document model `MLModelDocument`?

Reply:
DatasetDocument does the same and also uses the comma_pattern so that's why I had it in there. Doesn't seem to be necessary though, I'll switch it to one of the other analyzers and check if the comma analyzer is used properly anywhere.